python/benchmark/benchmark/bench_kmeans.py

```diff
@@ -185,11 +187,22 @@ def gpu_cache_df(df: DataFrame) -> DataFrame:
                 )
             elif is_vector_col:
                 df_for_scoring = transformed_df.select(
-                    vector_to_array(col(feature_col)), output_col
+                    vector_to_array(col(feature_col)).alias(feature_col), output_col
                 )
 
             cluster_centers = gpu_model.cluster_centers_
 
+            # temporary patch for DB with spark-rapids plugin
+            # this part is not timed so overhead is not critical, but should be reverted
+            # once https://github.com/NVIDIA/spark-rapids/issues/10770 is fixed
+            db_version = os.environ.get("DATABRICKS_RUNTIME_VERSION")
+            if db_version:
+                dim = len(cluster_centers[0])
+                # inject unsupported expr (slice) that is essentially a noop
+                df_for_scoring = df_for_scoring.select(
+                    F.slice(feature_col, 1, dim).alias(feature_col), output_col
```

Comment:
Interesting. Any intuition as to why slice can get the hanging resolved?

Reply:
Yes.  It is an unsupported expression in spark-rapids so falls back to cpu and injects columnartorow and rowtocolumnar transformations which do some batching that doesn't (but should after patching) happen otherwise.